packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +904,194 @@ export class AmplifyAuth
+          if (providerType === 'Facebook') {
+            outputProviders.push('FACEBOOK');
+          }
+          if (providerType === 'SignInWithApple') {
+            outputProviders.push('SIGN_IN_WITH_APPLE');
+          }
+          if (providerType === 'LoginWithAmazon') {
+            outputProviders.push('LOGIN_WITH_AMAZON');
+          }
+          if (providerType === 'OIDC') {
+            outputProviders.push(providerName);
+          }
+          if (providerType === 'SAML') {
+            outputProviders.push(providerName);
+          }
+        }
+        return JSON.stringify(outputProviders);
+      },
+    });
 
-    if (this.providerSetupResult.providersList.length > 0) {
-      output.socialProviders = JSON.stringify(
-        this.providerSetupResult.providersList
-      );
-    }
-    // if callback URLs are configured, we must expose the oauth settings to the output
-    if (
-      this.providerSetupResult.oAuthSettings &&
-      this.providerSetupResult.oAuthSettings.callbackUrls
-    ) {
-      const oAuthSettings = this.providerSetupResult.oAuthSettings;
-      if (this.domainPrefix) {
-        output.oauthCognitoDomain = `${this.domainPrefix}.auth.${
+    output.oauthCognitoDomain = Lazy.string({
+      produce: () => {
+        const userPoolDomain =
+          this.resources.userPool.node.tryFindChild('UserPoolDomain');
+        if (!userPoolDomain) {
+          return;
+        }
+        if (!(userPoolDomain instanceof UserPoolDomain)) {
+          throw Error('UserPoolDomain is not of type UserPoolDomain.');
+        }
+        return `${userPoolDomain.domainName}.auth.${
           Stack.of(this).region
         }.amazoncognito.com`;
-      }
+      },
+    });
 
-      output.oauthScope = JSON.stringify(
-        oAuthSettings.scopes?.map((s) => s.scopeName) ?? []
-      );
-      output.oauthRedirectSignIn = oAuthSettings.callbackUrls
-        ? oAuthSettings.callbackUrls.join(',')
-        : '';
-      output.oauthRedirectSignOut = oAuthSettings.logoutUrls
-        ? oAuthSettings.logoutUrls.join(',')
-        : '';
-      output.oauthClientId = this.resources.userPoolClient.userPoolClientId;
-      output.oauthResponseType = 'code';
-    }
+    output.oauthScope = Lazy.string({
+      produce: () => {
+        return JSON.stringify(cfnUserPoolClient.allowedOAuthScopes ?? []);
+      },
+    });
+
+    output.oauthRedirectSignIn = Lazy.string({
+      produce: () => {
+        return cfnUserPoolClient.callbackUrLs
+          ? cfnUserPoolClient.callbackUrLs.join(',')
+          : '';
+      },
+    });
+
+    output.oauthRedirectSignOut = Lazy.string({
+      produce: () => {
+        return cfnUserPoolClient.logoutUrLs
+          ? cfnUserPoolClient.logoutUrLs.join(',')
+          : '';
+      },
+    });
+
+    output.oauthResponseType = Lazy.string({
+      produce: () => {
+        return cfnUserPoolClient.allowedOAuthFlows
+          ? cfnUserPoolClient.allowedOAuthFlows.join(',')
+          : '';
```

Comment:
Should the default be `code` as it was before?

Reply:
the default is set above and should be returned in the left hand side of the ternary operation
<img width="889" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/c9ee4747-32c9-43da-ab94-9efa1c191fee">